src/header.rs

```diff
@@ -240,11 +250,28 @@ impl Header {
         }
     }
 
+    /// Returns the size of file's data this header represents.
+    ///
+    /// This is different from `Header::size` for sparse files, which have
+    /// some longer `size()` but shorter `data_size()`.
+    ///
+    /// May return an error if the field is corrupted.
+    pub fn data_size(&self) -> io::Result<u64> {
```

Comment:
Perhaps this could be something like `entry_size`? It's just the number of bytes following the header, right?